pkgs/checks/lib/src/checks.dart

```diff
@@ -375,9 +376,9 @@ class _TestContext<T> implements Context<T>, _ClauseDescription {
     required void Function(CheckFailure) fail,
     required bool allowAsync,
     required bool allowUnawaited,
-    String? label,
+    Iterable<String> Function()? label,
   })  : _value = value,
-        _label = label ?? '',
+        _label = label ?? (() => ['']),
```

Comment:
An empty Iterable would not work for the root context. I realized when I started to write the doc that it would be better to keep the old signature which enforces it explicitly, and always wrap the string in a closure here.

For the other constructor it would work to use an empty iterable because it is never read, so I added a static method to tear off instead of creating a closure.